tests/graph/test_networks.py

```diff
@@ -60,3 +64,88 @@ def test_initialize_without_category(self, request: pytest.FixtureRequest):
+        _polygon = Polygon([(0., 0.), (0., 1.), (1., 1.), (1., 0.), (0., 0.)])
+        _link_type = ""
+        _network_type = ""
+        with pytest.raises(ValueError) as exc_err:
+            Network.get_clean_graph_from_osm_download(_polygon, _link_type, _network_type)
+
+        assert str(exc_err.value) == 'Unrecognized network_type ""'
+
+    @slow_test
+    def test_get_clean_graph_from_osm_download_for_output_type(self):
+        _test_input_directory = Path(r'C:\repos\ra2ce\tests\test_data\graph\test_networks')
+        _polygon_file = _test_input_directory / "_test_polygon.geojson"
+        poly_dict = nut.read_geojson(geojson_file=_polygon_file)
+        _link_type = ""
+        _network_type = "drive"
+        graph_complex = Network.get_clean_graph_from_osm_download(
+            polygon=nut.geojson_to_shp(poly_dict),
+            network_type=_network_type,
+            link_type=_link_type
+        )
+        assert isinstance(graph_complex, Graph) == True
+
+    @slow_test
+    def test_get_clean_graph_from_osm_download_for_specific_node(self):
+        _test_input_directory = Path(r'C:\repos\ra2ce\tests\test_data\graph\test_networks')
+        _polygon_file = _test_input_directory / "_test_polygon.geojson"
+        poly_dict = nut.read_geojson(geojson_file=_polygon_file)
+        _link_type = ""
+        _network_type = "drive"
+        graph_complex = Network.get_clean_graph_from_osm_download(
+            polygon=nut.geojson_to_shp(poly_dict),
+            network_type=_network_type,
+            link_type=_link_type
+        )
+
+        # reference: https://www.openstreetmap.org/node/1402598729#map=17/51.98816/4.39126&layers=T
+        _osm_id_node_to_validate = 4987298323
+        if _osm_id_node_to_validate in list(graph_complex.nodes.keys()):
+            node_to_validate_exists = True
+        else:
+            node_to_validate_exists = False
+
+        assert node_to_validate_exists == True
+
+    @slow_test
+    def test_get_clean_graph_from_osm_download_for_specific_edge(self):
+        _test_input_directory = Path(r'C:\repos\ra2ce\tests\test_data\graph\test_networks')
+        _polygon_file = _test_input_directory / "_test_polygon.geojson"
+        poly_dict = nut.read_geojson(geojson_file=_polygon_file)
+        _link_type = ""
+        _network_type = "drive"
+        graph_complex = Network.get_clean_graph_from_osm_download(
+            polygon=nut.geojson_to_shp(poly_dict),
+            network_type=_network_type,
+            link_type=_link_type
+        )
+        # reference: https://www.openstreetmap.org/way/334316041#map=19/51.98945/4.39166&layers=T
+        _osm_id_edge_to_validate = 334316041
+        for edge_attribute in list(graph_complex.edges.values()):
+            if edge_attribute["osmid"] == _osm_id_edge_to_validate:
+                _osm_id_edge_to_validate = True
+                break
+        assert _osm_id_edge_to_validate == True
```

Comment:
This can be simplified:
```suggestion
        assert any(edge_attribute["osmid"] == _osm_id_edge_to_validate for edge_attribute in list(graph_complex.edges.values()))
```

Or even better:
```suggestion
        assert _osm_id_edge_to_validate in list(map(lambda x: x["osmid"]), graph_complex.edges.values())
```